### PR TITLE
Opensource LA and histogram kernels

### PR DESCRIPTION
**Pull request by artem-shaporenko**
No build procedure currently

Signed-off-by: Artem Shaporenko <artem.shaporenko@intel.com>

### CODE REVIEW

**Comment by dvrogozh** [comment]
@artem-shaporenko : tests have failed. Can you, please, look into that?

**Comment by artem-shaporenko** [comment]
> @artem-shaporenko : tests have failed. Can you, please, look into that?

Not related to this commit - broken before, need just rebase - will do

File: _studio/mfx_lib/genx/h264_encode/src/genx_histogram.cpp [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    SurfaceIndex OUTBUF,
+    uint max_h_pos,
+    uint max_v_pos,
+    uint off_h,
+    uint off_v)
+{
+    const uint SLM_SIZE_INT = 256;
+    const uint SLM_SIZE = SLM_SIZE_INT * sizeof(uint);
+
+    vector<uint, 64> vUI64;
```
Comment (lakulako) [inline _studio/mfx_lib/genx/h264_encode/src/genx_histogram.cpp]: Would be nice to have includes for vector, matrix etc. here

File: _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp [hunk 2]
```diff
@@ -1,15 +1,15 @@
-// Copyright (c) 2017 Intel Corporation
-// 
+// Copyright (c) 2018 Intel Corporation
```
[the remark this answers is not in the document]
Comment (artem-shaporenko) [reply _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: Doesn't make any sense

File: _studio/mfx_lib/genx/h264_encode/src/genx_simple_me.cpp [hunk 3]
```diff
@@ -0,0 +1,2825 @@
...
+    // set up parameters
+    MaxMvLen[0] = 512;
+    MaxMvLen[1] = maxMvLenY / 4;
+    RefSize[0] = PicSize[1] * 16;
+    RefSize[1] = (PicSize[0] + 1) * 16;
+
+    /* fields and MBAFF are not supported */
+
+    // remove quater pixel fraction
+    Predictor >>= 2;
```
[the remark this answers is not in the document]
Comment (artem-shaporenko) [reply _studio/mfx_lib/genx/h264_encode/src/genx_simple_me.cpp]: Yes

File: _studio/mfx_lib/genx/h264_encode/src/genx_simple_me.cpp [hunk 4]
```diff
@@ -0,0 +1,2825 @@
...
+
+    //
+    // set the reference position
+    //
+
+    Ref = Predictor;
+    Ref[1] &= -2;
```
[the remark this answers is not in the document]
Comment (artem-shaporenko) [reply _studio/mfx_lib/genx/h264_encode/src/genx_simple_me.cpp]: God suggestion, thank you, next time

File: _studio/mfx_lib/genx/h264_encode/src/genx_simple_me.cpp [hunk 5]
```diff
@@ -0,0 +1,2825 @@
...
+{
+    uchar best_uniOut_MbSubShape;
+    matrix<uchar, 11, 32> best_imeOut;
+
+    best_imeOut.select<4, 1, 32, 1> (7, 0) = best_imeOutIn;
+
+    if (GET_CURBE_AllFractional(CURBEData))
+    {
+        U8 mode;
+        uchar FBRMbMode, FBRSubMbShape, FBRSubPredMode;
```
[the remark this answers is not in the document]
Comment (artem-shaporenko) [reply _studio/mfx_lib/genx/h264_encode/src/genx_simple_me.cpp]: Historically

File: _studio/mfx_lib/genx/h264_encode/src/genx_histogram.cpp (hunk 1 above)
Comment (artem-shaporenko) [reply _studio/mfx_lib/genx/h264_encode/src/genx_histogram.cpp]: They are in MDF package for compiler, you are not including std into your code right?

Comment (dmitry-gurulev) [reply _studio/mfx_lib/genx/h264_encode/src/genx_histogram.cpp]: @lakulako, they are defined at <cm_vm.h> that is included by <cm/cm.h>

File: _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (lakulako) [reply _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: It is just a rule.

[the remark this answers is not in the document]
Comment (artem-shaporenko) [reply _studio/mfx_lib/fei/h264_la/mfx_h264_la.cpp]: Don't know such rule, can say more - it is like this in most of files, for example you can take a look here:
https://github.com/Intel-Media-SDK/MediaSDK/blob/ea817cbbfc76d1bff78fd180f02ee4dbc4ac063c/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp#L1

**Pull request merged**